fltkhs.cabal

```diff
@@ -12,29 +12,62 @@ bug-reports: https://github.com/deech/fltkhs/issues
 copyright: © 2017 Aditya Siram All Rights Reserved
 category: UI,Graphics,GUI,User Interfaces
 build-type: Custom
-cabal-version: >=1.24
+cabal-version: >=2.0
+  -- for custom-setup 
 source-repository head
   type: git
   location: http://github.com/deech/fltkhs
 
-Flag Bundled
+----------------------------------------
+-- by default, 
+-- 
+-- * no OpenGL support
+-- * the system `fltk` package is used (not the bundled tarball)
+--
+
+Flag bundled
    Description: Use the bundled FLTK library.
    Manual: True
    Default: False
 
-Flag Opengl
+Flag opengl
    Description: Enable OpenGL support.
    Manual : True
    Default: False
 
+Flag executables
+   Description: Build executables.
+   Manual : True
+   Default: True
+
+----------------------------------------
+
 custom-setup
   setup-depends:
     base >= 4.5 && < 4.11,
     Cabal >= 1.24.1 && < 1.25,
     filepath,
     directory >= 1.2.3.0
 
+----------------------------------------
+
 library
+
+  extra-libraries: jpeg
+      -- from the `libjpeg` system package 
+
+  build-tools: autoconf, c2hs
+
+  if !flag(bundled) 
+     -- when `bundled`, we don't need a system `fltk`, since we build it ourself
+     extra-libraries: fltk
+     build-tools: fltk-config
+        -- the `fltk` system package provides the `fltk-config` executable
+
+  if flag(opengl)
+     extra-libraries: GL, GLU
+         -- e.g. from the `mesa` system package 
```

Comment:
In the past I've had a lot of issues getting the `extra-libraries` field to correctly detect the presence of a C library and unfortunately it still doesn't work (at least on my Debian box). With `fltkhs`, `fltk-config` has been a much better way, I use the `configure` script to populate `fltkhs.buildinfo`. It probably screws up `cabal2nix` but I'm inclined to remove these conditionals because it seems to break the build. Sorry.

Reply:
No worries, building reliably is definitely more important. 

Can we leave those lines in the cabal file, commented out? Or in the contributing.md, with an explanation? It'd be nice to include that information somewhere besides these issues, if someone else, in the future, wants to try it.